File: frontend/src/lib/config/deployment.ts
/**
 * Deployment mode for the application
 * @enum {string}
 */
export type DeploymentMode = 'demo' | 'production';

/**
 * Configuration for demo mode features
 */
export interface DemoConfig {
  /** Artificial delay for API responses (in milliseconds) */
  apiDelay: number;
  /** Probability of simulated transaction failure (0.0 - 1.0) */
  failureRate: number;
  /** Maximum allowed transaction amount in demo mode */
  maxTransactionAmount: number;
}

/**
 * Get the current deployment mode from environment
 * @returns {DeploymentMode} Current deployment mode
 */
export const getDeploymentMode = (): DeploymentMode => {
  return (process.env.NEXT_PUBLIC_DEPLOYMENT_MODE === 'demo' ? 'demo' : 'production');
};

/**
 * Check if the application is running in demo mode
 * @returns {boolean} True if running in demo mode
 */
export const isDemoMode = (): boolean => {
  return getDeploymentMode() === 'demo';
};

/**
 * Get demo mode configuration
 * Only relevant when running in demo mode
 * @returns {DemoConfig} Demo mode configuration
 */
export const getDemoConfig = (): DemoConfig => {
  return {
    apiDelay: parseInt(process.env.NEXT_PUBLIC_MOCK_API_DELAY || '2000', 10),
    failureRate: parseFloat(process.env.NEXT_PUBLIC_MOCK_FAILURE_RATE || '0.1'),
    maxTransactionAmount: parseFloat(process.env.NEXT_PUBLIC_DEMO_MAX_TRANSACTION || '10000'),
  };
};

/**
 * Feature flags configuration based on deployment mode
 */
export const features = {
  /** Core features - always enabled */
  enableCore: true,
  
  /** Demo-only features */
  enableMockApi: isDemoMode(),
  enableTestData: isDemoMode(),
  
  /** Production-only features */
  enableLiveApi: !isDemoMode(),
  enableRealTransactions: !isDemoMode(),
  
  /** MVP limitations */
  enableStaking: false,
  enableGovernance: false,
  enableLending: false,
  
  /** Configurable features */
  enableKyc: process.env.NEXT_PUBLIC_ENABLE_KYC === 'true',
  enableTransactions: process.env.NEXT_PUBLIC_ENABLE_TRANSACTIONS === 'true',
};

// Log deployment mode on app initialization
if (typeof window !== 'undefined') {
  const mode = getDeploymentMode();
  console.log(`🚀 NEDApay Wallet running in ${mode} mode`);
  
  if (isDemoMode()) {
    console.log('⚠️ Demo Mode - Limited features enabled');
    console.log('Features:', features);
  }
}

File: src/config/deployment.rs
use std::env;

/// Represents the deployment mode of the application
#[derive(Debug, Clone, PartialEq)]
pub enum DeploymentMode {
    /// Demo mode for bank partner demonstrations
    /// - Uses sandbox APIs
    /// - Restricts certain features
    /// - Shows mock data
    Demo,
    
    /// Production mode for real users
    /// - Uses live APIs
    /// - Full feature access
    /// - Real transaction processing
    Production,
}

impl DeploymentMode {
    /// Returns true if running in demo mode
    pub fn is_demo(&self) -> bool {
        matches!(self, DeploymentMode::Demo)
    }

    /// Returns true if running in production mode
    pub fn is_production(&self) -> bool {
        matches!(self, DeploymentMode::Production)
    }
}

impl Default for DeploymentMode {
    fn default() -> Self {
        DeploymentMode::Production
    }
}

impl From<String> for DeploymentMode {
    fn from(s: String) -> Self {
        match s.to_lowercase().as_str() {
            "demo" => DeploymentMode::Demo,
            _ => DeploymentMode::Production,
        }
    }
}

/// Get the current deployment mode from environment
/// 
/// # Returns
/// - `DeploymentMode::Demo` if DEPLOYMENT_MODE=demo
/// - `DeploymentMode::Production` otherwise
/// 
/// # Example
/// ```rust
/// use crate::config::deployment::get_deployment_mode;
/// 
/// let mode = get_deployment_mode();
/// if mode.is_demo() {
///     println!("Running in demo mode!");
/// }
/// ```
pub fn get_deployment_mode() -> DeploymentMode {
    env::var("DEPLOYMENT_MODE")
        .map(DeploymentMode::from)
        .unwrap_or_default()
}

/// Configuration for demo mode features
#[derive(Debug, Clone)]
pub struct DemoConfig {
    /// Artificial delay for API responses (in milliseconds)
    pub api_delay: u64,
    
    /// Probability of simulated transaction failure (0.0 - 1.0)
    pub failure_rate: f64,
    
    /// Maximum allowed transaction amount in demo mode
    pub max_transaction_amount: f64,
}

impl Default for DemoConfig {
    fn default() -> Self {
        Self {
            api_delay: 2000,
            failure_rate: 0.1,
            max_transaction_amount: 10000.0,
        }
    }
}

/// Get demo mode configuration
/// Only relevant when running in demo mode
pub fn get_demo_config() -> DemoConfig {
    DemoConfig {
        api_delay: env::var("DEMO_API_DELAY")
            .ok()
            .and_then(|v| v.parse().ok())
            .unwrap_or(2000),
        
        failure_rate: env::var("DEMO_FAILURE_RATE")
            .ok()
            .and_then(|v| v.parse().ok())
            .unwrap_or(0.1),
            
        max_transaction_amount: env::var("DEMO_MAX_TRANSACTION")
            .ok()
            .and_then(|v| v.parse().ok())
            .unwrap_or(10000.0),
    }
}

#[cfg(test)]
mod tests {
    use super::*;
    
    #[test]
    fn test_deployment_mode_from_string() {
        assert!(matches!(DeploymentMode::from("demo".to_string()), DeploymentMode::Demo));
        assert!(matches!(DeploymentMode::from("DEMO".to_string()), DeploymentMode::Demo));
        assert!(matches!(DeploymentMode::from("production".to_string()), DeploymentMode::Production));
        assert!(matches!(DeploymentMode::from("invalid".to_string()), DeploymentMode::Production));
    }
    
    #[test]
    fn test_deployment_mode_helpers() {
        let demo = DeploymentMode::Demo;
        let prod = DeploymentMode::Production;
        
        assert!(demo.is_demo());
        assert!(!demo.is_production());
        assert!(!prod.is_demo());
        assert!(prod.is_production());
    }
}
